apps/arweave/src/ar_poa.erl

```diff
@@ -26,11 +26,16 @@ validate(Args) ->
 				RecallOffset - BlockStartOffset
 		end,
 	ValidateDataPathFun =
-		case BlockStartOffset >= StrictDataSplitThreshold of
+		case BlockStartOffset >= MerkleRebaseSupportThreshold of
```

Comment:
What do you think about moving this code block and the one in [ar_data_sync.erl](https://github.com/ArweaveTeam/arweave/pull/433/files#diff-2934ddad2ed46ac194a6be7abde4be6d440bc7565e1128bba234fad9fd2bd5d2R929) into ar_merkle.erl?

Reply:
Makes total sense, although I would move it to ar_poa and keep ar_merkle the place to define basic merkle tree functionality without the specifics of the protocol like rule-upgrading thresholds.